### PR TITLE
1.8 Changes (Rebase)

### PR DESCRIPTION
**Pull request by ribagi**
See #82 No other changes so far that have not be listed there.

### CODE REVIEW

File: config.yml [hunk 1]
```diff
@@ -779,6 +945,265 @@ production_factories:
...
+        amount: 64   
+    recipes:
+      - Create_Leather_Leggings
+      - Create_Leather_Boots
+      - Create_Leather_Chestplate
+      - Create_Leather_Helmet
+    repair_multiple: 2
+    repair_inputs:
+      Sticks:
+        material: STICK
```
Comment (ProgrammerDan) [inline config.yml]: Is the material to fix the leather armor factory supposed to be "sticks"?

File: config.yml [hunk 2]
```diff
@@ -779,6 +945,265 @@ production_factories:
...
+        material: COAL
+        durability: 1
+    inputs:  
+      Iron Blocks:
+        material: IRON_BLOCK
+        amount: 96   
+    recipes:
+      - Create_Iron_Axe
+      - Create_Iron_Pickaxe
+      - Create_Iron_Spade
```
Comment (ProgrammerDan) [inline config.yml]: Hoes are excluded from random tool factory?

I'm guessing shears are intentionally left in the Iron Forge, yes?

File: config.yml [hunk 3]
```diff
@@ -779,6 +945,265 @@ production_factories:
...
+        durability: 1
+    inputs:  
+      Sticks:
+        material: STICK
+        amount: 96   
+    recipes:
+      - Create_Wooden_Axe
+      - Create_Wooden_Spade
+      - Create_Wooden_Axe
+      - Create_Fishing_Rod
```
Comment (ProgrammerDan) [inline config.yml]: No Wooden hoes?

I'm guessing it's just a lack of enchantment so no need for them to be here, correct?

File: config.yml [hunk 4]
```diff
@@ -779,6 +945,265 @@ production_factories:
...
+    fuel: 
+      Charcoal:
+        material: COAL
+        durability: 1
+    inputs:
+      Emerald Blocks:
+        material: EMERALD_BLOCK
+        amount: 32  
+    recipes:
+      - Create_Bow
```
Comment (ProgrammerDan) [inline config.yml]: I love the inclusion of Bow here. Bravo.

File: config.yml [hunk 5]
```diff
@@ -2097,7 +2522,7 @@ production_factories:
       - Forge_Iron_Bars
       - Forge_Buckets
       - Forge_Iron_Doors
-      - Forge_Iron_Trapdoors
+
```
Comment (ProgrammerDan) [inline config.yml]: Was the removal of Iron Trapdoors here intentional?

File: config.yml [hunk 6]
```diff
@@ -4560,8 +4905,8 @@ production_recipes:
     production_time: 32
     inputs:
       Clay:
-        material: CLAY
-        amount: 128
+        material: CLAY_BALL
+        amount: 512
```
Comment (ProgrammerDan) [inline config.yml]: Pretty sure this is wrong. We moved to clay blocks for these recipes intentionally

File: config.yml [hunk 7]
```diff
@@ -4571,8 +4916,8 @@ production_recipes:
     production_time: 5
     inputs:
       Clay:
-        material: CLAY
-        amount: 16
+        material: CLAY_BALL
+        amount: 64
```
Comment (ProgrammerDan) [inline config.yml]: Same note, this is a regression error. Should be clay, not balls.

File: config.yml [hunk 8]
```diff
@@ -779,6 +945,265 @@ production_factories:
...
+      Diamond Block:
+        material: DIAMOND_BLOCK
+        amount: 32
+      Emerald Block:
+        material: EMERALD_BLOCK
+        amount: 64
+    recipes:
+      - Enchant_Sharpness
+      - Enchant_Fire_Aspect
+      - Enchant_Looting
```
Comment (ProgrammerDan) [inline config.yml]: Bane of Anthropods is p. important for animal enthusiasts, I'd recommend it be added here as an achievable recipe.

File: config.yml [hunk 9]
```diff
@@ -7723,3 +8036,2703 @@ production_recipes:
...
+        level: 3
+        probability: 0.16
+      Efficiency 4:
+        type: DIG_SPEED
+        level: 4
+        probability: 0.16
+      Efficiency 4:
+        type: DIG_SPEED
+        level: 4
+        probability: 0.06
```
Comment (ProgrammerDan) [inline config.yml]: You have Efficiency 4 three times here; I'm guessing this was a TODO: to replace with new enchantment types but was never done?

File: config.yml [hunk 10]
```diff
@@ -7723,3 +8036,2703 @@ production_recipes:
...
+        level: 2
+        probability: 0.007
+      Smite 5:
+        type: DAMAGE_UNDEAD
+        level: 5
+        probability: 0.012
+      Bane of Arthropods 4:
+        type: DAMAGE_ARTHROPODS
+        level: 4
+        probability: 0.0105
```
Comment (ProgrammerDan) [inline config.yml]: Bane lvl 4 appears twice, given probability I'm guessing this was meant to be Bane 5.

Note that my actual realized presence of B5 in normal enchanting was closer to 0.04, but my sample size was only maybe a hundred swords.

File: config.yml [hunk 11]
```diff
@@ -7723,3 +8036,2703 @@ production_recipes:
...
+      Projectile Protection 4:
+        type: PROTECTION_PROJECTILE
+        level: 4
+        probability: 0.27
+      Protection 4:
+        type: PROTECTION_ENVIRONMENTAL
+        level: 4
+        probability: 0.03
+      Throns 2:
+        type: THRONS
```
Comment (ProgrammerDan) [inline config.yml]: Intentional typo? Does THRONS evaluate to THORNS?

File: config.yml [hunk 12]
```diff
@@ -7723,3 +8036,2703 @@ production_recipes:
...
+        level: 2
+        probability: 0.007
+      Smite 5:
+        type: DAMAGE_UNDEAD
+        level: 5
+        probability: 0.012
+      Bane of Arthropods 4:
+        type: DAMAGE_ARTHROPODS
+        level: 4
+        probability: 0.0105
```
Comment (ProgrammerDan) [inline config.yml]: Similar to before, pretty sure you meant Bane 5. (duplicate otherwise)

File: config.yml [hunk 13]
```diff
@@ -7723,3 +8036,2703 @@ production_recipes:
...
```
Comment (ProgrammerDan) [inline config.yml]: Same comment here, Bane 4 appears twice.

**Comment by ProgrammerDan** [comment]
Finally did a full read through of the configuration, made a few notations. Mostly seems clean, but there are a few problems, including a regression on the clay recipes.

**Comment by ttk2** [comment]
these need to be fixed if they are in fact mistakes, thanks Dan

**Comment by ribagi** [comment]
They are a mistake. I am going to fix them now.

**Comment by ribagi** [comment]
Looking at some of these, I got to say that my original repo must have been in a bad spot because I fixed many of these issues before.

**Comment by ribagi** [comment]
I just tried to look up the enchantments on Hoes, end up there are no enchantments for hoes. No wounder why matta and I never added them in.

**Comment by ProgrammerDan** [comment]
I figured that was likely the case

**Comment by MonsieurWTF** [comment]
It should be noted that the outputs of the Iron Cauldron are 36 XP bottles (4 Emeralds), and the outputs of the Wood Cauldrons are 4 Emeralds.

Other than what type of materials are put in, the two cauldrons are on equal production value. In fact, the ingredients for the Wood Cauldron are even harder to acquire than Iron Cauldron.

 Wood_XP_1:
+    name: Wood XP Factory 2
 +    production_time: 80
+    inputs:
+      Endstone:
+        material: ENDER_STONE
 +        amount: 16
+      Nether Warts:
+        material: NETHER_WARTS
 +        amount: 256
+      Glass Bottle:
+        material: GLASS_BOTTLE
 +        amount: 36
+    outputs:
+      Emerald:
+        material: EMERALD
 +        amount: 4

End Stone?

**Comment by ribagi** [comment]
I see the issue. I didn't change bottoles to emeralds. Gonna fix this tonight.

**Comment by ribagi** [comment]
Fix an issue where De/Compactor was not working.

**Comment by ttk2** [comment]
Should I update the config or the jar?

**Comment by ProgrammerDan** [comment]
Quick checkpoint: Have all the specific issues I raised been either addressed or dismissed?

File: config.yml [hunk 14]
```diff
@@ -648,6 +678,135 @@ production_factories:
...
+        amount: 256   
+    recipes:
+      - Emerald_XP_0
+      - Emerald_XP_1
+      - Emerald_XP_2
+    repair_multiple: 2
+    repair_inputs:
+      Emeralds:
+        material: EMERALD
+        amount: 32
```
Comment (ProgrammerDan) [inline config.yml]: Convention check: 10% of install would be approx 19 Emeralds, vs. 32 listed here.

File: config.yml [hunk 15]
```diff
@@ -648,6 +678,135 @@ production_factories:
...
+      - Nether_Decraft_Crate_of_Spruce_Sapling
+      - Nether_Decraft_Crate_of_Sugar_Cane
+      - Nether_Decraft_Crate_of_Vine
+      - Nether_Decraft_Crate_of_Wheat
+      - Nether_Decraft_Crate_of_Yellow_Flower
+    repair_multiple: 2
+    repair_inputs:
+      Diamonds:
+        material: DIAMOND
+        amount: 12
```
Comment (ProgrammerDan) [inline config.yml]: Convention check: 10% of the install cost in diamonds would be 6, not 12.

**Comment by ttk2** [comment]
seconding dan's question

File: config.yml [hunk 16]
```diff
@@ -7723,3 +8034,2725 @@ production_recipes:
...
+        material: EMERALD
+        amount: 4
+  Wood_XP_2:
+    name: Wood XP Factory 3
+    production_time: 80
+    inputs:
+      Carrot:
+        material: CARROT
+        amount: 640
+      Wheat:
```
Comment (ProgrammerDan) [inline config.yml]: Just a label issue: Wheat should be Soul Sand on this line.

File: config.yml [hunk 17]
```diff
@@ -6218,17 +6540,6 @@ production_recipes:
       Iron Door:
         material: IRON_DOOR
         amount: 18
-  Forge_Iron_Trapdoors:
```
Comment (ProgrammerDan) [inline config.yml]: This removal is still here; still a regression. Iron Trapdoors are live on the server now, I doubt removing them is intentional.

**Comment by ProgrammerDan** [comment]
@ribagi Revised my comments based on latest commit. Down to four issues; one regression, two inconsistent repair costs, and one mislabel. Great stuff!

**Comment by MonsieurWTF** [comment]
That not signing the Contrib. License Agreement.

File: config.yml [hunk 18]
```diff
@@ -79,22 +103,34 @@ repair_factory:
...
+        material: 'COAL_ORE'
+        amount: 3
     recipe:
-      AAA_DIAMOND:
-        material: 'DIAMOND_BLOCK'
-        amount: 1
+      Coal Ore:
+        material: 'COAL_ORE'
+        amount: 64
+	  Emerald Blocks:
```
Comment (ProgrammerDan) [inline config.yml]: Tabs vs. Spaces: This will not parse as YAML

File: config.yml [hunk 19]
```diff
@@ -648,6 +681,135 @@ production_factories:
...
+        material: IRON_INGOT
+        amount: 256   
+    recipes:
+      - Emerald_XP_0
+      - Emerald_XP_1
+      - Emerald_XP_2
+    repair_multiple: 10
+    repair_inputs:
+      Emeralds:
+        material: EMERALD
```
Comment (ProgrammerDan) [inline config.yml]: Emerald vs. Emerald_Block -- Amount seems right iif material = EMERALD_BLOCK

File: config.yml [hunk 20]
```diff
@@ -648,6 +681,135 @@ production_factories:
...
+        amount: 2
+      Diamonds:
+        material: DIAMOND
+        amount: 2  
+      Iron:
+        material: IRON_INGOT
+        amount: 25
```
Comment (ProgrammerDan) [inline config.yml]: for repair multiple of 10, to maintain 10% ratio, amount should be 2; unless you want it to cost install cost in Iron Ingots to repair.

Comment (ttk2) [inline config.yml]: How's progress on these?

File: config.yml [hunk 21]
```diff
@@ -7723,3 +8037,2727 @@ production_recipes:
...
+        material: DIAMOND_CHESTPLATE
+        amount: 5
+      Emerald Block:
+        material: EMERALD_BLOCK
+        amount: 44
+    outputs:
+      Diamond Chestplate:
+        material: DIAMOND_CHESTPLATE
+        amount: 5
+    enchantments:
```
Comment (MonsieurWTF) [inline config.yml]: I'm going to guess that this is supposed to be Unbreaking and not Protection

File: config.yml [hunk 22]
```diff
@@ -7723,3 +8037,2727 @@ production_recipes:
...
+      Emerald Block:
+        material: EMERALD_BLOCK
+        amount: 42
+    outputs:
+      Diamond Legs:
+        material: DIAMOND_LEGGINGS 
+        amount: 5
+    enchantments:
+      Protection:
+        type: DURABILITY
```
Comment (MonsieurWTF) [inline config.yml]: Same here.

File: config.yml [hunk 23]
```diff
@@ -7723,3 +8037,2727 @@ production_recipes:
...
+      Emerald Block:
+        material: EMERALD_BLOCK
+        amount: 36
+    outputs:
+      Diamond Boots:
+        material: DIAMOND_BOOTS  
+        amount: 5
+    enchantments:
+      Protection:
+        type: DURABILITY
```
Comment (MonsieurWTF) [inline config.yml]: And here.

File: config.yml [hunk 24]
```diff
@@ -7723,3 +8037,2727 @@ production_recipes:
...
+      Emerald Block:
+        material: EMERALD_BLOCK
+        amount: 32
+    outputs:
+      Diamond Helemt:
+        material: DIAMOND_HELMET   
+        amount: 5
+    enchantments:
+      Protection:
+        type: DURABILITY
```
Comment (MonsieurWTF) [inline config.yml]: .

File: config.yml [hunk 25]
```diff
@@ -7723,3 +8037,2727 @@ production_recipes:
...
+      Emerald Block:
+        material: EMERALD_BLOCK
+        amount: 36
+    outputs:
+      Diamond Pick:
+        material: DIAMOND_PICKAXE   
+        amount: 5
+    enchantments:
+      Protection:
+        type: DIG_SPEED
```
Comment (MonsieurWTF) [inline config.yml]: .

File: config.yml [hunk 26]
```diff
@@ -7723,3 +8037,2727 @@ production_recipes:
...
+      Emerald Block:
+        material: EMERALD_BLOCK
+        amount: 36
+    outputs:
+      Diamond Spade:
+        material: DIAMOND_SPADE   
+        amount: 5
+    enchantments:
+      Protection:
+        type: DIG_SPEED
```
Comment (MonsieurWTF) [inline config.yml]: Protection from dig speed is no joke.

File: config.yml [hunk 27]
```diff
@@ -7723,3 +8037,2727 @@ production_recipes:
...
+      Emerald Block:
+        material: EMERALD_BLOCK
+        amount: 75
+    outputs:
+      Diamond Pick:
+        material: DIAMOND_PICKAXE  
+        amount: 5
+    enchantments:
+      Protection:
+        type: SILK_TOUCH
```
Comment (MonsieurWTF) [inline config.yml]: More Protection changing.

File: config.yml [hunk 28]
```diff
@@ -7723,3 +8037,2727 @@ production_recipes:
...
+      Emerald Block:
+        material: EMERALD_BLOCK
+        amount: 75
+    outputs:
+      Diamond Spade:
+        material: DIAMOND_SPADE   
+        amount: 5
+    enchantments:
+      Protection:
+        type: SILK_TOUCH
```
Comment (MonsieurWTF) [inline config.yml]: .

File: config.yml [hunk 29]
```diff
@@ -7723,3 +8037,2727 @@ production_recipes:
...
+        material: EMERALD_BLOCK
+        amount: 75
+    outputs:
+      Diamond Pick:
+        material: DIAMOND_AXE  
+        amount: 5
+    enchantments:
+      Protection:
+        type: SILK_TOUCH
+        level: 1
```
Comment (MonsieurWTF) [inline config.yml]: .

**Comment by ttk2** [comment]
the config in this form fails to validate, also someone needs to sign the CLA

**Comment by MonsieurWTF** [comment]
@Goldmattress Still needs to sign, and I think Dan said that there's tabs inside the config.

**Comment by ProgrammerDan** [comment]
@MonsieurWTF Yeah, basically I'm just going to start issuing pull requests to fix problems, my inline comments are not being addressed in a consistent fashion. It's messing with my OCD :P

**Comment by MonsieurWTF** [comment]
You can criticize mine if you want. #90 . I'd like to work out any kinks in it if there are any while you're around and reading.

**Comment by ProgrammerDan** [comment]
It'd be great if a second pair of eyes could scan the config in depth, but I am not seeing any other errors at this time.

**Comment by ttk2** [comment]
Nice. Check that stack trace I reported this morning. I get the feeling
it's the io changes and not the config but I want to be sure.

**Comment by DoctorCelebro** [comment]
Not sure if intentional but the diamond cauldrons require 256 bottles when the output is only 162 bottles (2 emerald blocks).

**Comment by ProgrammerDan** [comment]
@DoctorCelebro Good catch; at the moment, the 94 bottle different is probably easiest to correct after the fact. 

I think in discussions with Matta he mentioned that "glass" -- or sand -- was a bit of a special case, so this might be intentional. @Goldmattress any thoughts?

**Comment by ProgrammerDan** [comment]
Merge conflicts 100% resolved:

Was a fucking messy rebase, wound up having to squash all these commits to make it work, and a bunch of other tl;dr, but it's done now and available here: https://github.com/Civcraft/FactoryMod/pull/98

**Pull request closed without merging**